acs-engine.go

```diff
@@ -107,6 +108,46 @@ func saveFile(dir string, file string, data []byte) error {
 	return nil
 }
 
+// injectCACertificateAndKey reads the ca cert and key files if they exist and inserts them into the container service model
+func injectCACertificateAndKey(containerService *api.ContainerService, caCertificatePath string, caKeyPath string) {
```

Comment:
This seems overly complicated:

1. Why do you length check here instead of at the call site? I think this function should only be called if the args are specified. It seems like it's abusing `len(string)==0` to represent a nil value or a case where the function shouldn't be called. See the other comment below about the flag.

2. `ioutil.Readfile` will handle the error if the file is missing.

3. You could pass a `*string` into the function instead of the entire containerService model and have it set the string value. That way you can un-duplicate the body of this function and just call it twice.